File: ContactsManagerAppWithJavaROOMDBMVVM/app/src/main/java/com/masterclass/contactsmanagerappwithjavaroomdbmvvm/activityClickHandlers/AddContactsClickHandler.java
package com.masterclass.contactsmanagerappwithjavaroomdbmvvm.activityClickHandlers;

import android.content.Context;
import android.content.Intent;
import android.view.View;
import android.widget.Toast;

import com.masterclass.contactsmanagerappwithjavaroomdbmvvm.activities.MainActivity;
import com.masterclass.contactsmanagerappwithjavaroomdbmvvm.database.Contacts;
import com.masterclass.contactsmanagerappwithjavaroomdbmvvm.viewModels.MyViewModel;

public class AddContactsClickHandler {

    Contacts contact;
    Context context;
    MyViewModel myViewModel;

    public AddContactsClickHandler(Contacts contact, Context context, MyViewModel myViewModel) {
        this.contact = contact;
        this.context = context;
        this.myViewModel = myViewModel;
    }

    public void onSubmitButtonClicked(View view){
        if (contact.getName() == null || contact.getPhoneNumber() == null || contact.getEmail() == null){
            Toast.makeText(context, "All fields are required", Toast.LENGTH_SHORT).show();
        }
        else{
            Intent i = new Intent(context, MainActivity.class);

            Contacts c = new Contacts(contact.getName(), contact.getPhoneNumber(), contact.getEmail());
            myViewModel.addNewContact(c);

            context.startActivity(i);
        }
    }
}

File: ContactsManagerAppWithKotlinROOMDBMVVM2/app/src/main/java/com/masterclass/contactsmanagerappwithkotlinroomdbmvvm/activityClickHandlers/AddContactsClickHandler.kt
package com.masterclass.contactsmanagerappwithkotlinroomdbmvvm.activityClickHandlers

import android.content.Context
import android.content.Intent
import android.view.View
import android.widget.Toast
import com.masterclass.contactsmanagerappwithkotlinroomdbmvvm.activities.MainActivity
import com.masterclass.contactsmanagerappwithkotlinroomdbmvvm.database.Contacts
import com.masterclass.contactsmanagerappwithkotlinroomdbmvvm.viewModels.MyViewModel
import kotlin.random.Random


class AddContactsClickHandler(
    private val contact: Contacts,
    private val context: Context,
    private val myViewModel: MyViewModel
) {

    fun onSubmitButtonClicked(view: View) {
        if (contact.name == null || contact.phoneNumber == null || contact.email == null) {
            Toast.makeText(context, "All fields are required", Toast.LENGTH_SHORT).show()
        } else {
            val intent = Intent(context, MainActivity::class.java)
            val random = Random.Default
            val newContact = Contacts(null, contact.name, contact.phoneNumber, contact.email)
            myViewModel.addNewContact(newContact)

            context.startActivity(intent)
        }
    }
}
